beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -1101,6 +1122,12 @@ impl<T: BeaconChainTypes> ExecutionPendingBlock<T> {
         chain: &Arc<BeaconChain<T>>,
         notify_execution_layer: NotifyExecutionLayer,
     ) -> Result<Self, BlockError<T::EthSpec>> {
+        chain
```

Comment:
@jmcph4  Sorry for the late ping here. Stumbled into this while reviewing some deneb stuff.
Wondering why we observe the block here. We would hit this condition from multiple paths:
1. When verifying a chain segment during range sync: we attempt to observe proposals from before the finalized slot which would be redundant. 
2. For verification of RPC blocks: doesn't really make sense to observe RPC blocks in a cache for blocks received over gossip

Reply:
@pawanjay176 We've kind of expanded the scope of the cache beyond just gossip, because the relays need to detect _all_ equivocations, regardless of whether the block is fetched via gossip or RPC.

I agree that observing finalized blocks is not useful though, so we should probably refactor to only observe blocks within a small window of the head block.

Is this causing problems on Deneb?

Reply:
Ahh gotcha. No issues on deneb, was just a bit confusing to me.
We should probably also change the [docs](https://github.com/sigp/lighthouse/blob/unstable/beacon_node/beacon_chain/src/observed_block_producers.rs#L1-L2) to reflect the scope if we are refactoring to only observe in a small distance